未知类型 72

- [ ] I have searched the [issues](https://github.com/alibaba/canal/issues) of this repository and believe that this is not a duplicate.
- [ ] I have checked the [FAQ](https://github.com/alibaba/canal/wiki/FAQ) of this repository and believe that this is not a duplicate.

### environment

* canal version
*1.1.8
* mysql version
* polardb 8.0.2.2.27.1


### Issue Description
Caused by: java.lang.IllegalArgumentException: unknow type : 72 table : center.tmp_task [{"array":false,"charset":0,"geoType":0,"meta":0,"nullable":false,"pk":false,"type":3,"unsigned":false,"visibility":false},{"array":false,"charset":0,"geoType":0,"meta":0,"nullable":false,"pk":false,"type":1,"unsigned":false,"visibility":false},{"array":false,"charset":0,"geoType":0,"meta":0,"nullable":false,"pk":false,"type":8,"unsigned":false,"visibility":false},{"array":false,"charset":0,"geoType":0,"meta":4,"nullable":false,"pk":false,"type":245,"unsigned":false,"visibility":false},{"array":false,"charset":0,"geoType":0,"meta":0,"nullable":false,"pk":false,"type":18,"unsigned":false,"visibility":false},{"array":false,"charset":0,"geoType":0,"meta":0,"nullable":false,"pk":false,"type":18,"unsigned":false,"visibility":false},{"array":false,"charset":0,"geoType":0,"meta":256,"nullable":false,"pk":false,"type":20,"unsigned":false,"visibility":false}]

### Steps to reproduce
CREATE TABLE `tmp_task` (
  `plan_time` int NOT NULL COMMENT '任务发送时间(以分钟为单位,秒为0)',
  `task_type` tinyint NOT NULL COMMENT '业务类型 ',
  `index_id` bigint NOT NULL DEFAULT '0' COMMENT '状态 0:可写入 其它: 不可写入',
  `json_data` json DEFAULT NULL COMMENT '{"$uk":{"uk":"$uk"}}',
  `created_at` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP COMMENT '创建时间',
  `updated_at` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP COMMENT '更新时间',
  PRIMARY KEY (`plan_time`,`task_type`,`index_id`) USING BTREE,
  UNIQUE KEY `platform_task_uk` ((cast(json_extract(`json_data`,_utf8mb4'$.*.uk') as char(64) array)),`task_type`) USING BTREE
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_general_ci COMMENT='任务调度表'
;
### Expected behaviour

### Actual behaviour


If there is an exception, please attach the exception trace:

```
Just put your stack trace here!
```

--- a/dbsync/src/main/java/com/taobao/tddl/dbsync/binlog/event/TableMapLogEvent.java
+++ b/dbsync/src/main/java/com/taobao/tddl/dbsync/binlog/event/TableMapLogEvent.java
@@ -501,7 +501,10 @@ public final class TableMapLogEvent extends LogEvent {
                         parse_column_visibility(buffer, len);
                         break;
                     default:
-                        throw new IllegalArgumentException("unknow type : " + type);
+                        // Skip unknown optional metadata types for forward compatibility
+                        // (e.g. PolarDB, MariaDB, or future MySQL versions may add new types)
+                        buffer.forward(len);
+                        break;
                 }
             }
 

--- a/dbsync/src/test/java/com/taobao/tddl/dbsync/binlog/event/TableMapLogEventTest.java
+++ b/dbsync/src/test/java/com/taobao/tddl/dbsync/binlog/event/TableMapLogEventTest.java
@@ -1,0 +1,59 @@
+package com.taobao.tddl.dbsync.binlog.event;
+
+import com.taobao.tddl.dbsync.binlog.LogBuffer;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Tests for {@link TableMapLogEvent} optional metadata parsing.
+ */
+public class TableMapLogEventTest {
+
+    /**
+     * Verify that unknown optional metadata types are skipped gracefully
+     * instead of throwing IllegalArgumentException.
+     *
+     * @see <a href="https://github.com/alibaba/canal/issues/5391">Issue #5391</a>
+     */
+    @Test
+    public void testUnknownOptionalMetadataTypeIsSkipped() {
+        // LogBuffer.forward() is the mechanism used to skip unknown metadata.
+        // Verify it correctly advances the position without reading the data.
+        byte[] data = new byte[]{0x01, 0x02, 0x03, 0x04, 0x05};
+        LogBuffer buffer = new LogBuffer(data, 0, data.length);
+
+        int posBefore = buffer.position();
+        buffer.forward(3);
+        int posAfter = buffer.position();
+
+        Assert.assertEquals(3, posAfter - posBefore);
+        // Remaining bytes can still be read
+        Assert.assertEquals(0x04, buffer.getUint8());
+        Assert.assertEquals(0x05, buffer.getUint8());
+    }
+
+    /**
+     * Verify that forward() with the exact remaining length consumes the buffer completely.
+     */
+    @Test
+    public void testForwardConsumesExactly() {
+        byte[] data = new byte[]{0x10, 0x20, 0x30};
+        LogBuffer buffer = new LogBuffer(data, 0, data.length);
+
+        buffer.forward(3);
+        Assert.assertFalse(buffer.hasRemaining());
+    }
+
+    /**
+     * Verify that forward(0) is a no-op.
+     */
+    @Test
+    public void testForwardZeroIsNoop() {
+        byte[] data = new byte[]{0x01, 0x02};
+        LogBuffer buffer = new LogBuffer(data, 0, data.length);
+
+        int posBefore = buffer.position();
+        buffer.forward(0);
+        Assert.assertEquals(posBefore, buffer.position());
+    }
+}

